src/main/java/org/folio/clients/SynchronousOkapiClient.java

```diff
@@ -0,0 +1,63 @@
+package org.folio.clients;
+
+import io.vertx.core.json.JsonObject;
+import org.apache.http.HttpHeaders;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.folio.util.OkapiConnectionParams;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.core.MediaType;
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.folio.rest.RestVerticle.OKAPI_HEADER_TENANT;
+import static org.folio.rest.RestVerticle.OKAPI_HEADER_TOKEN;
+
+/**
+ * Base class for okapi clients that use blocking method for http calls.
+ */
+public abstract class SynchronousOkapiClient {
+
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  protected final OkapiConnectionParams okapiConnectionParams;
+  private final CloseableHttpClient httpClient;
+
+  public SynchronousOkapiClient(OkapiConnectionParams okapiConnectionParams) {
+    this.okapiConnectionParams = okapiConnectionParams;
+    httpClient = HttpClients.createDefault();
+  }
+
+  public Optional<JsonObject> getByIds(List<String> ids) {
+    HttpGet httpGet = new HttpGet();
+    setCommonHeaders(httpGet);
+    prepareRequest(httpGet, ids);
+
+    try (CloseableHttpResponse response = httpClient.execute(httpGet)) {
+      return Optional.ofNullable(postProcess(response));
+    } catch (IOException e) {
+      log.error("Exception while calling " + httpGet.getURI(), e);
+      return Optional.empty();
+    }
```

Comment:
should be closing the response , it is leaking connections
check: https://stackoverflow.com/questions/19920037/why-does-me-use-httpclients-createdefault-as-httpclient-singleton-instance-exe

Reply:
The response is being closed, it's in try with resources block.

Reply:
If there is a need to close the client, I think, it should be handled elsewhere.

Reply:
We definitely need to close the client.  What is elsewhere?

Reply:
Closing it In the calling code would be my suggestion.

Reply:
ok I went through the code again, seems fair in this situation

Reply:
Lets make to check this when we write that code